### PR TITLE
Output and record standard streams in in-toto-run

### LINKED ISSUE #11: Properly duplicate standard streams
Currently we provide a command line flag to `toto-run` to either don't touch stdout and stderr, which is required if `toto-run` wraps interactive commands like e.g. `vi`, or to redirect and store them as byproducts to the link metadata.
Ideally we would like to duplicate the streams to allow both.
- Using a pipe won't work because processes like vi will complain
- Wrapping stdout/sterr in Python won't work because the suprocess will only take the filedescriptor and then uses it natively
- Reading from /dev/(stdout|stderr), /dev/tty is *NIX specific

### PR DESCRIPTION
**Pull request by lukpueh**
**Fixes issue #**:
Fixes #11 

**Description of the changes being introduced by the pull request**:
Add `subprocess.run`-like function to `in_toto.process` module that redirects standard output and standard error of the executed command to temporary files and both writes the contents to the corresponding streams of the parent process during execution and also returns them along
with the child process's return code upon its termination.

This function is called `in_toto_run` when passing `record_streams=True`.

The PR is a friendly takeover of #160. Thanks for the solid ground work, @ng1355! 

**Please verify and check that the pull request fulfills the following
requirements**:

- [x] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [x] Tests have been added for the bug fix or new feature
- [x] Docs have been added for the bug fix or new feature

### CODE REVIEW

File: in_toto/process.py [hunk 1]
```diff
@@ -40,11 +45,11 @@
 PIPE = subprocess.PIPE
 
 
+
```
Comment (SantiagoTorres) [inline in_toto/process.py]: Probably this stray newline can go away.

File: in_toto/process.py [hunk 2]
```diff
@@ -118,3 +123,97 @@ def run(cmd, check=True, timeout=SUBPROCESS_TIMEOUT, **kwargs):
...
+          raise subprocess.TimeoutExpired(cmd, timeout)
+
+        # Read from child process's redirected streams, write to parent
+        # process's standard streams and construct retuirn values
+        stdout_part = stdout_reader.read()
+        stderr_part = stderr_reader.read()
+        sys.stdout.write(stdout_part)
+        sys.stderr.write(stderr_part)
+        sys.stdout.flush()
+        sys.stderr.flush()
```
Comment (SantiagoTorres) [inline in_toto/process.py:210]: silly comment, but are all of these IPC calls available on windows too?

File: tests/test_process.py [hunk 3]
```diff
@@ -50,5 +54,67 @@ def test_input_vs_stdin(self):
...
+
+    for cmd in [cmd_str, cmd_list]:
+      return_code, _, _ = in_toto.process.run_duplicate_streams(cmd)
+      self.assertEqual(return_code, 100)
+
+
+  def test_run_duplicate_streams_timeout(self):
+    """Test raise TimeoutExpired. """
+    with self.assertRaises(in_toto.process.subprocess.TimeoutExpired):
+      in_toto.process.run_duplicate_streams("python --version", timeout=-1)
```
Comment (SantiagoTorres) [inline tests/test_process.py:117]: how is this timing out?

File: in_toto/process.py (hunk 2 above)
Comment (lukpueh) [reply in_toto/process.py:210]: According to the passing windows tests on appveyor they do.

File: tests/test_process.py (hunk 3 above)
Comment (lukpueh) [reply tests/test_process.py:117]: It times out a second before it started, and thus returns immediately. `subprocess`' timeout has the same behavior.

**Pull request merged**